### ISSUE TITLE
Some confusions about Eagle3 training

### ISSUE THREAD

**Issue by ShallowMDream** [sgl-project/SpecForge#128]
I used Qwen2.5-7B as the target model and the default LlamaForCausalLMEagle3 as the draft model. After training on 50K samples from ShareGPT for 5 epochs, I ran run_gsm8k.py, achieving an accuracy of 0.83 and an average accepted token length of 1.94.

There is a significant gap compared to the accepted token length reported in the paper. What should I do to optimize it? Should I use more training data, or create a more suitable draft model?

**Comment by FrankLeeeee** [comment]
Hi @ShallowMDream , increasing the data will help. The paper used more than 500k data samples while sharegpt in our codebase contains only 120k.

**Comment by ShallowMDream** [comment]
> Hi [@ShallowMDream](https://github.com/ShallowMDream) , increasing the data will help. The paper used more than 500k data samples while sharegpt in our codebase contains only 120k.

Thanks, @FrankLeeeee . 
Thank you for your suggestion. I will try to use more data for training.

**Comment by zyksir** [comment]
@ShallowMDream The paper use 68k in sharegpt and 200k in ultrachat, you can try this first
